src/Network/IHTTPRequest.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * @copyright Copyright (C) 2020, Friendica
+ *
+ * @license GNU APGL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Network;
+
+/**
+ * Interface for calling HTTP requests and returning their responses
+ */
+interface IHTTPRequest
```

Comment:
What will this interface be used for besides `Network\HTTPRequest`?

Reply:
It's part of the "SOLID" principle:
https://en.wikipedia.org/wiki/Interface_segregation_principle

Concrete benefit/usage (for this place, but equal for every class): I'll add a "guzzle" way of `IHTTPRequest`. Now it's possible to switch between "curl" way and "guzzle" way and f.e. test if the result is the same.

Reply:
In this case, can we identify the first class you created as being Curl instead of a generic `HTTPRequest`?

Reply:
Jep `CurlHTTPRequest` is maybe a good name for this class?

Reply:
This brings the question of the return type of `IHTTPRequest->get`, is it supposed to be a `CurlResult` in all cases?

Reply:
I was already altering the result to a general `ResponseInterface` but this is far more invasiv and so I chose the step by step variant :)
The next step is to altering our curl code to return this interface :)